### ISSUE TITLE
Suggested elections in Admin view

### ISSUE THREAD

**Issue by pmk01** [DemocracyClub/EveryElection#1051]
Suggested elections seem to create an ID in admin view even if rejected. Anyway, the following are in Admin but aren't standalone by-elections:

local.bolton.astley-bridge.by.2021-05-06
local.harlow.church-langley.by.2021-05-06
local.harlow.toddbrook.by.2021-05-06
local.sefton.blundellsands.by.2021-05-06
local.southend-on-sea.st-lukes.by.2021-05-06
local.southend-on-sea.westborough.by.2021-05-06
local.wakefield.airedale-and-ferry-fryston.by.2021-05-06

Possibly no issue at all, but thought I'd raise it.

**Comment by michaeljcollinsuk** [comment]
@symroe when attempting to delete these in the admin ([I actually tried with a EU election from 1998](https://elections.democracyclub.org.uk/admin/elections/election/32457/change/)), I've encountered a permissions error:
![Screenshot 2021-06-10 at 16 09 50](https://user-images.githubusercontent.com/15347726/121550080-533c3000-ca06-11eb-8333-5a4dd3d20d92.png)

I've tried giving myself moderation history related permissions via the admin, but still get the same thing. Just wanted to check if this might be by design, as a failsafe?

**Comment by symroe** [comment]
See:

https://github.com/DemocracyClub/EveryElection/blob/master/every_election/apps/elections/admin.py#L120-L128
and
https://github.com/DemocracyClub/EveryElection/wiki/Cancelled-Elections-and-Soft-Deletes

But I think that elections that have been rejected in moderation are a special case to this.

Maybe we want to change the logic (somewhere?) to deal with the case where there's a single moderation status (or two, one for "suggested" and one "rejected") and allow deleting in that instance? We'll need to think through how best to do that, though!

**Comment by chris48s** [comment]
What do you gain by deleting an election where the latest moderation status is rejected?
If it is "clutter" in the admin, is the solution to hide rejected elections where the latest status is not Approved in the places you don't want to see them?

If it actually causes an issue, another approach we considered at one point is creating suggested elections with a "provisional" id when they are suggested then they only get assigned a "proper" id at the point they are approved. That is probably cleaner as then you don't "pollute" the namespace of theoretical identifiers with incorrect suggestions.

**Comment by michaeljcollinsuk** [comment]
I do think really it is about clutter - so I would probably agree that hiding rejected suggestions in the admin could be the way to go (and is very straightforward. What do you think @pmk01? Am I missing other reasons than cluttering the admin?

**Comment by pmk01** [comment]
yes I think concealing them would be fine. One thought would be elections which are rejected in error... would this solution have any effect if we tried to re-create them?

**Comment by chris48s** [comment]
It is super uncommon but occasionally it is necessary to take a rejected or deleted election and approve it (this is done by manually adding an Approved `ModerationStatus` record via the admin).

Probably the most sensible solution is to set it so that the default filter in admin hides the elections you don't want to see but you can show deleted/suggested/rejected if you need to. Having records that do exist in your DB but are _completely invisible_ via /admin creates more problems than it solves :)

This is one of the reasons there are two managers for the Election object - via the front-end you mostly want to hide everything other than approved, but doing that _everywhere_ breaks things in surprising ways:
https://github.com/DemocracyClub/EveryElection/blob/7cb862a62e97e4fca2556e5aba8fbeb511292271/every_election/apps/elections/managers.py#L76-L99
https://github.com/DemocracyClub/EveryElection/blob/7cb862a62e97e4fca2556e5aba8fbeb511292271/every_election/apps/elections/models.py#L230-L251

**Comment by GeoWill** [comment]
+1 from me to not having things in the DB which are invisible in the admin - from the principle that we should move in the direction of reducing technical barriers to working with EE. A filter which hides by default sounds good.
